ci/lint/Dockerfile

```diff
@@ -0,0 +1,19 @@
+# See test/lint/README.md for usage.
```

Comment:
nit: Given that this isn't used by the ci, mabye move this file to `test/lint` and remove this comment? (The paths below would need to be adjusted)

Reply:
Yeah I started this way, but because the docker build context either (i) pulls in entire tree when run from root (super slow) and (ii) doesn't allow "back references" to files outside of the context, it kind of has to live here. There wasn't another good workaround that I could find.

Reply:
nit: Ah ok, maybe add a comment why this needs to live here.